service-mirror.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apiError "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+func main() {
+	// Set up logger .
+	log := &logrus.Logger{
+		Out:   os.Stderr,
+		Level: logrus.DebugLevel,
+		Formatter: &logrus.TextFormatter{
+			TimestampFormat: "2006-01-02 15:04:05",
+			ForceColors:     true,
+			DisableColors:   false,
+			FullTimestamp:   true,
+		},
+	}
+	//Shows line number: Too long
+	// log.SetReportCaller(true)
+
+	log.Info("-- Starting Global Mirror ---")
+
+	var kubeconfig *string
+	if home := homedir.HomeDir(); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
```

Comment:
is this so that we can run it out of the cluster?

Reply:
We won't need to pass this in a pod, so perhaps it's just best to default it to "".

Reply:
Yeah like while testing it out of cluster we need path to kubeconfig. When running in Pod i think we can use service accounts ?

Reply:
Updated it.

Reply:
Yup, service accounts should be picked up automatically so we're good.